packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -406,6 +406,24 @@ export class DocumentDeltaConnection
                 let isWebSocketTransportError = false;
                 try {
                     const description = error?.description;
+                    // const responseText = error?.context?.responseText;
+
+                    if (description.message?.includes("self signed certificate")) {
```

Comment:
Where does this error originate from? I.e. is it our service code, or some code that we do not control?
The better workflow would be not to do string matching, but rely on error codes / IDs, if possible.
Note - this code is shared across drivers.

You also assume that description is not undefined. I think you are better off moving this code into IF block below, which validates that description is valid object.

Reply:
1. Changed the initial logic of string-matching error message to relying on error code `DEPTH_ZERO_SELF_SIGNED_CERT`, which is accessible in `error.description` and `error.context`. This error code should be returned identically regardless of the browser the client is using. 

2. Updated the logic to first check whether `context` and `description` are undefined or not. We could concise the logic by making it `if ((description && typeof description === 'object' && description.error.code === 'DEPTH_ZERO_SELF_SIGNED_CERT`) || (context && typeof context=== 'object' && context.statusText.code === 'DEPTH_ZERO_SELF_SIGNED_CERT`))`. But do not think this would be too great for readability, so decided to break it to two pieces.